### PR TITLE
Basic Notification Support

### LINKED ISSUE #7: Add push notifications support
The mobile app integration has support for push notifications. This allows this app to register as a target. To do so, we should set `push_token` and `push_url` during the mobile app registration from #5 ([docs](https://developers.home-assistant.io/docs/en/app_integration_notifications.html)). We can use the same one as iOS.

@robbiet480 could you briefly explain how to get `push_token` and `push_url`?

### PR DESCRIPTION
**Pull request by JBassett**
This PR will do the following:
- Implement foreground receiving
- Push token and url to HA for notifications
- Update HA with new tokens

Fixes #7

### CODE REVIEW

**Comment by robbiet480** [comment]
🎉

File: domain/src/main/java/io/homeassistant/companion/android/domain/integration/IntegrationUseCase.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package io.homeassistant.companion.android.domain.integration
 interface IntegrationUseCase {
 
     suspend fun registerDevice(deviceRegistration: DeviceRegistration)
+    suspend fun updateRegistration(deviceRegistration: DeviceRegistration)
```
Comment (balloob) [inline domain/src/main/java/io/homeassistant/companion/android/domain/integration/IntegrationUseCase.kt]: Not all fields can be updated. From top of my head, I think we can't update encryption and maybe app id?

Comment (robbiet480) [reply domain/src/main/java/io/homeassistant/companion/android/domain/integration/IntegrationUseCase.kt]: [Correct](https://developers.home-assistant.io/docs/en/app_integration_sending_data.html#update-registration).

Only `app_data`, `app_version`, `device_name`, `manufacturer`, `model` and `os_version`.

Comment (JBassett) [reply domain/src/main/java/io/homeassistant/companion/android/domain/integration/IntegrationUseCase.kt]: I'll update the usecase to only have those fields, and reuse the objects on the backend side.

**Comment by andriej** [comment]
How can I activate those?

**Comment by dshokouhi** [comment]
@andriej restart HA to get the `notify.mobile_app_*` service call registered, also in the future hop on discord in #android we discuss these things there.

**Comment by dmaciaszek** [comment]
After restart HA I get `notify.mobile_app_*` as you suggested but when I call it by dev tools with payload `message: testing notification` nothing happen. In log I have error `Requested entity was not found.`. Am I doing something wrong?

**Comment by Zixim** [comment]
`notify.mobile_app_*`
Replace * with the correct device ?

**Comment by dmaciaszek** [comment]
Yes, selected from dropdown. Problem solved in https://github.com/home-assistant/home-assistant-android/issues/147

**Pull request merged**